core/drivers/tzc380.c

```diff
@@ -85,6 +85,31 @@ static void tzc_write_region_attributes(vaddr_t base, uint32_t region,
 	write32(val, base + REGION_ATTRIBUTES_OFF(region));
 }
 
+void tzc_set_region_size(uint8_t region, uint32_t size)
+{
+	uint32_t val;
+
+	val = tzc_read_region_attributes(tzc.base, region);
+	val &= ~(TZC_REGION_SIZE_MASK);
+	val |= TZC_ATTR_REGION_SIZE(size);
+	tzc_write_region_attributes(tzc.base, region, val);
+}
+
+void tzc_set_region_attr(uint8_t region, uint16_t attr)
+{
+	uint32_t val;
+
+	val = tzc_read_region_attributes(tzc.base, region);
+	val &= ~(TZC_ATTR_SP_MASK);
+
+	if (attr == (TZC_SP_NS_W | TZC_SP_NS_R | TZC_SP_S_W | TZC_SP_S_R))
+		val |= TZC_ATTR_SP_ALL;
+	if (attr == (TZC_SP_S_W | TZC_SP_S_R))
+		val |=  TZC_ATTR_SP_S_RW;
+
```

Comment:
what happens if user wants for instance NS R and S RW?

Reply:
Well, this could simply be extended to take care of these cases.

Reply:
yes sure that can be update , but you need to take care of:
all the several configuration options + take into account security inversion settings.